pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionAssignmentTailer.java

```diff
@@ -113,6 +112,11 @@ public synchronized void close() {
                     }
                 }
                 tailerThread = null;
+
+                // complete exit future to be safe
+                exitFuture.complete(null);
+                // reset the future
+                exitFuture = new CompletableFuture<>();
```

Comment:
What are you trying to do assign a completed future ?

Reply:
just reset it because this object can be close() and start() again